NEWS.md

```diff
@@ -1,203 +1,116 @@
+    propagating some of its options to 2nd and subsequent references of a parser
+    statement. This includes `drop-unmatched()`, `program-template()` and
+    `template()` options.
+    ([#3957](https://github.com/syslog-ng/syslog-ng/pull/3957))
+  * `match(), subst() and regexp-parser()`: fixed storing of numbered
+    (e.g.  $1,$2, $3 and so on) and named capture groups in regular expressions
+    in case the input of the regexp is the same as one of the match variables being
+    stored. In some cases the output of the regexp was clobbered and an invalid
+    value stored.
+    ([#3948](https://github.com/syslog-ng/syslog-ng/pull/3948))
+  * fix `threaded(no)` related crash: if threaded mode is disabled for
+    asynchronous sources and destinations (all syslog-like drivers such as
+    tcp/udp/syslog/network qualify), a use-after-free condition can happen due
+    to a reference counting bug in the non-threaded code path.  The
+    `threaded(yes)` setting has been the default since 3.6.1 so if you are using
+    a more recent version, you are most probably unaffected.  If you are using
+    `threaded(no)` a use-after-free condition happens as the connection closes.
+    The problem is more likely to surface on 32 bit platforms due to pointer
+    sizes and struct layouts where this causes a NULL pointer dereference.
+    ([#3997](https://github.com/syslog-ng/syslog-ng/pull/3997))
+
+## Packaging
+
+  * debian: fix logrotate file not doing the file rotation. (The path and command was invalid.)
+    ([#4031](https://github.com/syslog-ng/syslog-ng/pull/4031))
+  * `dbld`: add support for Fedora 35 in favour of Fedora 33
+    ([#3933](https://github.com/syslog-ng/syslog-ng/pull/3933))
 
 ## Other changes
 
- * The [syslog-ng Docker image](https://hub.docker.com/r/balabit/syslog-ng/)
-   is now automatically tagged and pushed to Docker Hub after each release
-   ([#3870](https://github.com/syslog-ng/syslog-ng/pull/3870))
- * `throttle()` filter: renamed to `rate-limit()`
-   ([#3866](https://github.com/syslog-ng/syslog-ng/pull/3866))
- * `python`: support Python 3.10
-   ([#3865](https://github.com/syslog-ng/syslog-ng/pull/3865))
- * `java`: upgraded from old log4j v1.x line to log4j v2.17.2
-   ([#3861](https://github.com/syslog-ng/syslog-ng/pull/3861))
-   ([#3927](https://github.com/syslog-ng/syslog-ng/pull/3927))
+  * `java()/python() destinations`: the `$SEQNUM` macro (and "seqnum" attribute in
+    Python) was erroneously for both local and non-local logs, while it should
+    have had a value only in case of local logs to match RFC5424 behavior
+    (section 7.3.1).  This bug is now fixed, but that means that all non-local
+    logs will have `$SEQNUM` set to zero from this version on, e.g.  the `$SEQNUM`
+    macro would expand to an string, to match the syslog() driver behaviour.
+    ([#3928](https://github.com/syslog-ng/syslog-ng/pull/3928))
+  * OpenSSL: add support for OpenSSL 3.0 (warning-free compilation)
```

Comment:
I would not mention the warning-free compilation.

Can we elaborate on the MD4 deprecation? I'd add a separate bullet point for it and mention that MD4 is deprecated and will be removed, and is no longer available when compiling with OpenSSL 3.0.